project/Build.scala

```diff
@@ -606,6 +610,8 @@ object Build {
         }
       },
 
+      concurrentRestrictions += Tags.limit(org.scalafmt.sbt.ConcurrentRestrictionTags.Scalafmt, 4),
```

Comment:
Limiting the parallelism of scalafmt seems to solve the problem of race conditions that previously led to a file being deleted. However, I'm a little concerned that this may only be mitigating the race condition instead of solving it. I think the solution would be to have the `scalafmt` commands in each subproject depend on the command in the previous subproject; we would go through subrpojects linearly, and scalafmt can still spawn threads for each task. What do you think, is there a better way to do that?

Reply:
I'll try to test what setting the concurrency limit to 1 does. Most likely, it solves the problem of race conditions.

Reply:
It did